src/main/host/managed_thread.rs

```diff
@@ -337,12 +327,41 @@ impl ManagedThread {
             other => panic!("Unexpected result from shim: {other:?}"),
         };
 
+        let native_pid = if flags & CloneFlags::CLONE_THREAD.bits() == 0 {
+            nix::unistd::Pid::from_raw(child_native_tid)
+        } else {
+            self.native_pid
+        };
+
+        if flags & CloneFlags::CLONE_THREAD.bits() == 0 {
```

Comment:
Convert flags to `CloneFlags` and use `flags.contains(CloneFlags::CLONE_THREAD)`? (And above)